spec-main/api-browser-window-spec.ts

```diff
@@ -4161,6 +4161,18 @@ describe('BrowserWindow module', () => {
         await leaveFullScreen;
         expect(w.isFullScreen()).to.be.false('isFullScreen');
       });
+
+      it('multiple windows inherit correct fullscreen state', async () => {
+        const w = new BrowserWindow();
+        const enterFullScreen = emittedOnce(w, 'enter-full-screen');
+        w.setFullScreen(true);
+        await enterFullScreen;
+        expect(w.isFullScreen()).to.be.true('isFullScreen');
+        await delay();
+        const w2 = new BrowserWindow();
+        await delay();
+        expect(w2.isFullScreen()).to.be.true('isFullScreen');
+      });
```

Comment:
should we add more tests for the other behaviors you mentioned?

Reply:
I couldn't find a good way to automate tests for scenarios 1 and 3. Scenario 1) requires to check that the Quick Panel from the OS receives the input event and Scenario 3) requires a custom window switcher like https://alt-tab-macos.netlify.app/ for example.